tests/test_transcription.py

```diff
@@ -0,0 +1,201 @@
+
+    for ref_f, est_f, sco_f in zip(ref_files, est_files, sco_files):
+        with open(sco_f, 'r') as f:
+            expected_scores = json.load(f)
+        # Load in reference transcription
+        ref_int, ref_pitch = mir_eval.io.load_valued_intervals(ref_f)
+        # Load in estimated transcription
+        est_int, est_pitch = mir_eval.io.load_valued_intervals(est_f)
+        scores = mir_eval.transcription.evaluate(ref_int, ref_pitch, est_int,
+                                                 est_pitch)
+        for metric in scores:
+            # This is a simple hack to make nosetest's messages more useful
+            yield (__check_score, scores[metric], expected_scores[metric])
+
+
+def test_invalid_pitch():
+
+    ref_int, ref_pitch = np.array([[0, 1]]), np.array([-100])
+    est_int, est_pitch = np.array([[0, 1]]), np.array([100])
+
+    yield raises(ValueError)(mir_eval.transcription.validate), \
+        ref_int, ref_pitch, est_int, est_pitch
+    yield raises(ValueError)(mir_eval.transcription.validate), \
+        est_int, est_pitch, ref_int, ref_pitch
+
+
+def test_inconsistent_int_pitch():
+
+    ref_int, ref_pitch = np.array([[0, 1], [2, 3]]), np.array([100])
+    est_int, est_pitch = np.array([[0, 1]]), np.array([100])
+
+    yield raises(ValueError)(mir_eval.transcription.validate), \
+        ref_int, ref_pitch, est_int, est_pitch
+    yield raises(ValueError)(mir_eval.transcription.validate), \
+        est_int, est_pitch, ref_int, ref_pitch
+
+
+def test_empty_ref():
+
+    warnings.resetwarnings()
+    warnings.simplefilter('always')
+    with warnings.catch_warnings(record=True) as out:
+
+        ref_int, ref_pitch = np.array([]), np.array([])
+        est_int, est_pitch = np.array([[0, 1]]), np.array([100])
+
+        mir_eval.transcription.validate(ref_int, ref_pitch, est_int, est_pitch)
+
+        # Make sure that the warning triggered
+        assert len(out) > 0
+
+        # And that the category is correct
+        assert out[0].category is UserWarning
+
+        # And that it says the right thing (roughly)
+        assert 'empty' in str(out[0].message).lower()
+
+
+def test_empty_est():
+
+    warnings.resetwarnings()
+    warnings.simplefilter('always')
+    with warnings.catch_warnings(record=True) as out:
+
+        ref_int, ref_pitch = np.array([[0, 1]]), np.array([100])
+        est_int, est_pitch = np.array([]), np.array([])
+
+        mir_eval.transcription.validate(ref_int, ref_pitch, est_int, est_pitch)
+
+        # Make sure that the warning triggered
+        assert len(out) > 0
+
+        # And that the category is correct
+        assert out[0].category is UserWarning
+
+        # And that it says the right thing (roughly)
+        assert 'empty' in str(out[0].message).lower()
+
+
+def test_precision_recall_f1_empty():
+
+    ref_int, ref_pitch = np.array([]), np.array([])
+    est_int, est_pitch = np.array([[0, 1]]), np.array([100])
+
+    precision, recall, f1 = \
```

Comment:
Nitpick: https://docs.python.org/2/howto/doanddont.html#using-backslash-to-continue-statements